tests/sync/partial_sync.cpp

```diff
@@ -231,11 +246,12 @@ TEST_CASE("Partial sync", "[sync]") {
         });
     }
 
-    SECTION("invalid queries") {
-        run_query("this isn't a valid query!", partial_config, PartialSyncTestObjects::A, [](Results, std::exception_ptr e) {
-            REQUIRE(e);
-        });
-    }
+//    TODO: Figure out how to test this.
```

Comment:
Actually testing all of the new convoluted logic for handling errors is very important.